src/gen-stochastic.js

```diff
@@ -406,4 +406,92 @@ class MarkovChain {
 		return c;
 	}
 }
-exports.MarkovChain = MarkovChain;
\ No newline at end of file
+
+class DeepMarkovChain {
+	constructor(data){
+		// transition probabilities table
+		this._table = new Map();
+		// train if dataset is provided
+		if (data) { this.train(data) };
+		// current state of markov chain
+		this._state = '';
+		// scoped random number generator
+		this.rng = seedrandom();
+	}
+	get table(){
+		// return copy of object
+		return this._table
+	}
+	clear(){
+		// empty the transition probabilities
+		this._table = new Map();
+	}
+	train(a, win){
+		if (!Array.isArray(a)){ 
+			return console.error('Error: train() expected array but received:', typeof a);
+		}
+		// build a transition table from array of values
+		for (let i=0; i < (a.length - win); i++) {
+			let slice = a.slice(i, i+win);
+			let key = JSON.stringify(slice);
+
+			let next = a[i+win]
+
+			if (this._table.has(key)) {
+				let arr = this._table.get(key)
+				arr.push(next)
+				this._table.set(key, arr)
+			} else {
+				this._table.set(key, [a[i+win]])
+			}
+		}
+	}
+	seed(s){
+		// set unpredictable seed if 0, null or undefined
+		if (s === 0 || s === null || s === undefined){
+			rng = seedrandom();
+		} else {
+			rng = seedrandom(s);
+		}
+	}
+	state(a){
+		// set the state
+		// stringify the state
+		let stringed = JSON.stringify(a)
+		if (!this._table.has(stringed)) {
+			console.error('Warning: state() value is not part of transition table');
+		}
+		this._state = stringed;
+	}
+	randomState() {
+		let keys = Array.from(this._table.keys())
+		this._state = keys[Math.floor(rng() * keys.length)]
```

Comment:
When the `randomState()` is used the `this._state` is now reset to the key that belongs to it, I'm not sure if this is desirable or if it should just choose a single number from possible output states.

eg. 
```js
let m = new Rand.DeepMarkovChain()

m.train([1, 2, 3, 4, 5, 6], 3);
m.state([1, 2, 3])
m.next()
// => 4 , state: [2, 3, 4]
m.next()
// => 5 , state: [3, 4, 5]
m.next()
// => 6 , state: [4, 5, 6]
m.next()
// => 4 , state: [2, 3, 4]
```
While I would think the new "random" state would be the new single value output appended to the previous states (eg. `[5, 6, 4]`), so maybe we need to keep track of a history, although this of course could result in a continuous run of impossible states, so not sure how to handle this.

Reply:
Your conclusion is why I went for the implementation that I did. It's up to you – I can code it where it maintains history or we can leave it as is.

Reply:
Yeah fair enough. I'll think a bit longer on how I like this to work.